### PR TITLE
Added additional logging for tests with unknown results.

### PR DESCRIPTION
**Pull request by AMZN-scspaldi**
Signed-off-by: scspaldi <scspaldi@amazon.com>

### CODE REVIEW

**Comment by AMZN-scspaldi** [comment]
Example output:

`=================== Test failure and error troubleshooting ====================
Use the following commands to re-run each test that failed locally
(NOTE: The 'PYTHON' or 'PYTHONPATH' environment variables need values for accurate commands): 
C:\Git\o3de\python\python.cmd -m pytest --build-directory C:/Git/o3de/build/windows/bin/profile  C:\Git\o3de\AutomatedTesting\Gem\PythonTests\editor_test_testing\TestSuite_Main.py::TestEditorTest::test_batched_1_crash_1_pass[windows-AutomatedTesting-windows_editor]

The following tests had an unknown result:

test_batched_1_crash_1_pass`

**Review by FuzzyCarterAWS — APPROVED** [review]
LGTM

File: Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/terminal_report.py [hunk 1]
```diff
@@ -67,6 +69,17 @@ def pytest_terminal_summary(terminalreporter, exitstatus, config):
                 "(NOTE: The 'PYTHON' or 'PYTHONPATH' environment variables need values for accurate commands): ",
                 test_path, nodeids, build_dir)
 
+            # Check for unknown test failures
+            have_printed_header = False
```
Comment (evanchia-ly-sdets) [inline Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/terminal_report.py:73]: I don't see where this gets switched on if the header already exists

**Review by evanchia-ly-sdets — CHANGES_REQUESTED** [review]

File: Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/terminal_report.py (hunk 1 above)
Comment (AMZN-scspaldi) [reply Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/terminal_report.py:73]: Good catch

**Review by evanchia-ly-sdets — APPROVED** [review]

**Pull request merged**